terraform/main/ssh/outputs.tf

```
@@ -1,13 +1,33 @@
 output "clusters" {
   value = {
     for i, cluster in local.clusters : cluster.name => {
-      local_name           = module.cluster[i].first_server_private_name
-      local_http_port      = module.cluster[i].local_http_port
-      local_https_port     = module.cluster[i].local_https_port
-      private_name         = module.cluster[i].first_server_private_name
-      kubeconfig           = module.cluster[i].kubeconfig
-      context              = module.cluster[i].context
+      kubeconfig = module.cluster[i].kubeconfig
+      context    = module.cluster[i].context
+
+      // alternative URL to reach the API from the same network this cluster is in
+      private_kubernetes_api_url = "https://${module.cluster[i].first_server_private_name}:6443"
+
+      // addresses of applications running in this cluster
+      app_addresses = {
+        public = { // resolvable over the Internet
+          name       = module.cluster[i].public_dns
```

public_dns is not defined in k3s output. Is it server_private_name?
```suggestion
          name       = module.cluster[i].first_server_private_name
```

---

Right, I did not check that well. Thanks for catching!